internal/workerutil/observability.go

```diff
@@ -75,3 +81,58 @@ func newOperations(observationContext *observation.Context, prefix string, keys,
 		handle: op("Handle"),
 	}
 }
+
+// lenientConcurrencyGaugeInterval is the interval at which a lenient
+// concurrency gauge's value is updated with the current window's maximum
+// value.
+const lenientConcurrencyGaugeInterval = time.Second * 5
+
+// newLenientConcurrencyGauge creates a new gauge-like object that emits
+// (into the given gauge) the maximum value over the last five seconds.
+// This method begins an immortal background routine.
+//
+// This gauge should be used to smooth-over the randomness sampled by
+// Prometheus by emitting the aggregate we'll likely be using with this
+// type of data directly.
+//
+// Without wrapping the numJobs gauge in this object, we tend to sample
+// zero handlers consistently when jobs are short-lived (< 500ms).
+// Keeping the max in memory gives us the value that we actually want
+// over time, and indicates an accurate level of concurrency.
+func newLenientConcurrencyGauge(gauge prometheus.Gauge) Gauge {
+	ch := make(chan float64)
+	go runLenientConcurrencyGauge(gauge, ch)
+
+	return &channelGauge{ch: ch}
+}
+
+func runLenientConcurrencyGauge(gauge prometheus.Gauge, ch <-chan float64) {
+	max := float64(0)
+	value := float64(0)
+	ticker := time.NewTicker(lenientConcurrencyGaugeInterval)
```

Comment:
It might be useful just to take the interval in the constructor.

```suggestion
func newLenientConcurrencyGauge(gauge prometheus.Gauge, interval time.Duration) Gauge {
	ch := make(chan float64)
	go runLenientConcurrencyGauge(gauge, ch, interval)

	return &channelGauge{ch: ch}
}

func runLenientConcurrencyGauge(gauge prometheus.Gauge, ch <-chan float64, interval time.Duration) {
	max := float64(0)
	value := float64(0)
	ticker := time.NewTicker(interval)
```